src/de/map.rs

```diff
@@ -605,6 +582,41 @@ where
             filter,
         })
     }
+
+    #[inline]
+    fn deserialize_struct<V>(
+        self,
+        name: &'static str,
+        fields: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        self.map.de.deserialize_struct(name, fields, visitor)
+    }
+
+    fn deserialize_enum<V>(
+        self,
+        _name: &'static str,
+        _variants: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        visitor.visit_enum(crate::de::var::EnumAccess::new(self.map.de))
+    }
```

Comment:
This call also incorrect, because it will use top-level deserializer for deserializing newtype variants, for example, but I leave fixing that for separate PR. We do not have tests for this situation yet, and I want to approve each change by tests that failed before and pass after the change.